ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/BlockWithReceipts.java

```diff
@@ -77,4 +73,27 @@ public String toString() {
         .add("receipts", receipts)
         .toString();
   }
+
+  public List<LogWithMetadata> getLogsWithMetadata(final boolean removed) {
```

Comment:
I'd move this logic to `LogsWithMetadata.create(Block block, List<Receipt> receipts, boolean removed)`.